Code/GraphMol/MolStandardize/Tautomer.h

```diff
@@ -35,8 +35,54 @@ typedef RDCatalog::HierarchCatalog<TautomerCatalogEntry, TautomerCatalogParams,
 namespace TautomerScoringFunctions {
 const std::string tautomerScoringVersion = "1.0.0";
 
+//! The SubstructTerm controls how Tautomers are generated
+///   Each Term is defined by a name, smarts pattern and score
+///   For example, the C=O term is defined as
+///    SubstructTerm("C=O", "[#6]=,:[#8]", 2)
+///   This gets a score of +2 for each Carbon doubly or aromatically
+///   Bonded to an Oxygen.
+///   For a list of current definitions, see getDefaultTautomerSubstructs
+struct RDKIT_MOLSTANDARDIZE_EXPORT SubstructTerm {
+  std::string name;
+  std::string smarts;
+  int score;
+  ROMOL_SPTR matcher;
```

Comment:
Using the static initializer remove the requirement for a flyweight.

I'm a little concerned here because matchers generally aren't thread safe which means that the tautomer scoring isn't thread safe.  n.b. this is probably ok without recursive queries. 

However,  we should probably just make these `ROMol matcher` and take the copy-construction hit to be thread-safe.

Reply:
Or mark it in the docs and show how to make these thread safe if necessary.

Reply:
> Using the static initializer remove the requirement for a flyweight.

Sure, but that means that you pay the cost of initializing everything upon library load, even if you never score a tautomer. The flyweight code is well tested and using it doesn't add much complexity, so what do we get for paying this price?
 
> I'm a little concerned here because matchers generally aren't thread safe which means that the tautomer scoring isn't thread safe. n.b. this is probably ok without recursive queries.
>
> However, we should probably just make these `ROMol matcher` and take the copy-construction hit to be thread-safe.

It's definitely ok without recursive queries, it's definitely a risk with recursive queries.

As long as we define a move ctor (or be explict about using the default one after making sure that it actually works) for the `SubstructTerm` and make sure we are always using references in the RDKit code, it seems like it's a good idea to switch to using ROMols instead of shared pointers

Reply:
I don't think a move c-tor would be appropriate here, we can't change the defaults.  If don't use ROMOL_SPTR the thread safe version would be:

```
std::vector<SubstructTerm> terms = getDefaultSubstructTerms(); // copy construct
scoreSubstructs(mol, terms);
```

We could just make a note of this.